assets/src/hooks/useDetour.ts

```diff
@@ -22,8 +22,12 @@ const useDetourDirections = (shapePoints: ShapePoint[]) =>
     }, [shapePoints]),
   })
 
-export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
-  const [snapshot, send] = useMachine(createDetourMachine)
+export const useDetour = (input: OriginalRoute) => {
```

Comment:
note:suggestion: I'd almost like to remove `OriginalRoute` here altogether and use, haven't come up with any downsides yet.
```ts
InputFrom<
  ActorLogicFrom<typeof createDetourMachine>
>
```

Reply:
This gives us the option to add more to `input` without having to change the `useDetour` interface to use it. going to add this to this PR.